private/buf/bufsync/syncer.go

```diff
@@ -56,33 +57,112 @@ func newSyncer(
 	return s, nil
 }
 
+// resolveSyncPoints resolves sync points for all known modules for the specified branch,
+// returning all modules for which sync points were found, along with their sync points.
+//
+// If a SyncPointResolver is not configured, this returns an empty map immediately.
+func (s *syncer) resolveSyncPoints(ctx context.Context, branch string) (map[Module]git.Hash, error) {
+	syncPoints := map[Module]git.Hash{}
+	// If resumption is not enabled, we can bail early.
+	if s.syncPointResolver == nil {
+		return syncPoints, nil
+	}
```

Comment:
Nit: Are the callers promised they they'll receive a not-nil map? Otherwise this looks a bit more clear:
```suggestion
	// If resumption is not enabled, we can bail early.
	if s.syncPointResolver == nil {
		return nil, nil
	}
	syncPoints := make(map[Module]git.Hash)
```

Reply:
Yes, they are promised a non-nil map, otherwise they'll have to nil-check it. I commented with that in the godoc.

aside: I dislike that the zero value for maps is useless while the zero value for slices is useful. Seems like an implementation detail that surfaces in the language semantics.